astroquery/mast/collections.py

```diff
@@ -0,0 +1,504 @@
+            one sepcific page of results.
+
+        Returns
+        --------
+        response : list of `~requests.Response`
+        """
+
+        self._current_connection = self._portal_api_connection
+
+        if isinstance(match, Row):
+            match = match["MatchID"]
+        match = str(match)  # np.int64 gives json serializer problems, so stringify right here
+
+        if version == 2:
+            service = "Mast.HscMatches.Db.v2"
+        else:
+            if version not in (3, None):
+                warnings.warn("Invalid HSC version number, defaulting to v3.", InputWarning)
+            service = "Mast.HscMatches.Db.v3"
+
+        params = {"input": match}
+
+        return self._current_connection.service_request_async(service, params, pagesize, page)
+
+    @class_or_instance
+    def get_hsc_spectra_async(self, pagesize=None, page=None):
+        """
+        Returns all Hubble Source Catalog spectra.
+
+        Parameters
+        ----------
+        pagesize : int, optional
+            Can be used to override the default pagesize.
+            E.g. when using a slow internet connection.
+        page : int, optional
+            Can be used to override the default behavior of all results being returned to obtain
+            one sepcific page of results.
+
+        Returns
+        --------
+        response : list of `~requests.Response`
+        """
+
+        self._current_connection = self._portal_api_connection
+
+        service = "Mast.HscSpectra.Db.All"
+        params = {}
+
+        return self._current_connection.service_request_async(service, params, pagesize, page)
+
+    def download_hsc_spectra(self, spectra, download_dir=None, cache=True, curl_flag=False):
+        """
+        Download one or more Hubble Source Catalog spectra.
+
+        Parameters
+        ----------
+        spectra : `~astropy.table.Table` or `~astropy.table.Row`
+            One or more HSC spectra to be downloaded.
+        download_dir : str, optional
+           Specify the base directory to download spectra into.
+           Spectra will be saved in the subdirectory download_dir/mastDownload/HSC.
+           If download_dir is not specified the base directory will be '.'.
+        cache : bool, optional
+            Default is True. If file is found on disc it will not be downloaded again.
+            Note: has no affect when downloading curl script.
+        curl_flag : bool, optional
+            Default is False.  If true instead of downloading files directly, a curl script
+            will be downloaded that can be used to download the data files at a later time.
+
+        Returns
+        --------
+        response : list of `~requests.Response`
+        """
+
+        # if spectra is not a Table, put it in a list
+        if isinstance(spectra, Row):
+            spectra = [spectra]
+
+        # set up the download directory and paths
+        if not download_dir:
```

Comment:
might be better to have `download_dir is not None` here.  `''` will result in a False evaluation here... that may be intentional, though?

Reply:
I definitely want `''` to also result in false evaluation. Would it be better to spell that out (`if download_dir is in (None, ''):`)?

Reply:
eh, it's fine as is.